### DISCUSSION TITLE
Ability to Import/Export from web database?

### DISCUSSION THREAD

**Discussion by velten-dev** [simolus3/drift#2875] [Q&A] [answered]
I was wondering what it would take to import/export data for the web WASM implementation of Drift. I was able to figure out how to export the encrypted database using a bit of Googling, but the web database is such a different creature, that I'm not sure where to begin to allow import export.

**Comment by simolus3** [comment]
Unfortunately, this is something that is not well supported yet. We don't have access to a uniform file system on the web. Instead, we have to pick the best thing that works in the current browser, which may be [something resembling a file system](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API) (but only available through a worker) or a custom file system stub built by storing chunks of blob data in IndexedDb. At the moment, there is no unified API to the file system layer on the web.

You can import an existing database the first time it's accessed with the `initializeDatabase` callback on [`WasmDatabase.open`](https://drift.simonbinder.eu/api/drift.wasm/wasmdatabase/open).

**Reply by simolus3** [reply, marked as answer]
https://github.com/simolus3/drift/issues/2777 is an existing issue tracking this.

**Reply by velten-dev** [reply]
Thank you for the reply. I will track the issue to keep tabs on this. 

I would say this is the one piece of functionality which would really bridge the gap between the native and web experience.
